core/trino-main/src/main/java/io/trino/security/AccessControlManager.java

```
@@ -862,112 +862,119 @@ public void checkCanSelectFromColumns(SecurityContext securityContext, Qualified
     }
 
     @Override
-    public void checkCanCreateRole(SecurityContext securityContext, String role, Optional<TrinoPrincipal> grantor, String catalogName)
+    public void checkCanCreateRole(SecurityContext securityContext, String role, Optional<TrinoPrincipal> grantor, Optional<String> catalogName)
     {
         requireNonNull(securityContext, "securityContext is null");
         requireNonNull(role, "role is null");
         requireNonNull(grantor, "grantor is null");
         requireNonNull(catalogName, "catalogName is null");
 
-        checkCanAccessCatalog(securityContext, catalogName);
-
-        catalogAuthorizationCheck(catalogName, securityContext, (control, context) -> control.checkCanCreateRole(context, role, grantor));
+        if (catalogName.isPresent()) {
```

If catalogName.isEmpty() we should either deny here or throw unsupported.